src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,228 @@
+        if max_std:
+            max_std = torch.Tensor([max_std])
+
+        if self._std_parameterization == 'exp':
+            self._init_std_param = init_std.log()
+            self._min_std_param = min_std.log() if min_std else None
+            self._max_std_param = max_std.log() if max_std else None
+
+        elif self._std_parameterization == 'softplus':
+            self._init_std_param = init_std.exp().add(-1).log()
+            self._min_std_param = min_std.exp().add(-1).log() if min_std else None
+            self._max_std_param = max_std.exp().add(-1).log() if max_std else None
+
+        else:
+            raise NotImplementedError
+
+        if std_share_network:
+            self.shared_mean_std_network = self._get_shared_mean_std_network()
+        else:
+            self.mean_module = self._get_mean_network()
+
+            if adaptive_std:
+                self.adaptive_std_module = self._get_adaptive_std_network()
+
+    def _get_shared_mean_std_network(self):
+        return MlpModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim * 2,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._init_shared_b,
+            layer_normalization=self._layer_normalization
+        )
+
+    def _init_shared_b(self, b):
+        return set_shared_b(
+                b[..., :self._action_dim], 0,
+                b[..., self._action_dim:], self._init_std_param.item())
+
+    def _get_mean_network(self):
+        return MlpModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+    def _get_adaptive_std_network(self):
+        return MlpModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._std_hidden_sizes,
+            hidden_nonlinearity=self._std_hidden_nonlinearity,
+            hidden_w_init=self._std_hidden_w_init,
+            hidden_b_init=self._std_hidden_b_init,
+            output_nonlinearity=self._std_output_nonlinearity,
+            output_w_init=self._std_output_w_init,
+            output_b_init=self._init_adaptive_b,
+            layer_normalization=self._layer_normalization)
+
+    def _init_adaptive_b(self, b):
+        return nn.init.constant_(b, self._init_std_param.item())
```

Comment:
Why not just use `nn. new_full`.

Reply:
I just tried to be similar with the nn.init methods (zeros_, ones_, ...)
(https://pytorch.org/docs/stable/_modules/torch/nn/init.html)

Do yo think using nn.new_full is better?

Reply:
Oh sorry. I think I made a mistake. `newfull` actually does not serve our purpose here. We need to be able to initialize b to arbitrary vector.